src/operator/tensor/broadcast_reduce-inl.cuh

```diff
@@ -60,16 +60,19 @@ __global__ void reduce_kernel(const int N, const int M, const bool addto,
           for (int u=0;u < unroll;u++) {
             idx_big[u] = idx_big0 + mxnet_op::unravel_dot(k + u*by, big_shape, big_stride);
           }
-          DType tmp[unroll];
+          AType tmp[unroll];
           #pragma unroll
           for (int u=0;u < unroll;u++) {
             if (k + u*by < Mend) {
               tmp[u] = OP::Map(big[idx_big[u]]);
+              // argmin/max, set IndexedNum.idx
+	      if (use_index)
+                *(reinterpret_cast<int*>(&tmp[u])) = k + u*by;
```

Comment:
That's pretty ugly and fragile. Maybe instead of `use_index` have a function template parameter here that by default would be a noop and you would change it to setting the index here?

Reply:
The plan was sort of to come back here and use c++ 17 `constexpr if` in the future when we support only cuda 11 and up. But yeah I have thought of this solution too and I will change it tomorrow

Reply:
I changed it